torch/onnx/symbolic_opset9.py

```diff
@@ -1982,15 +1982,18 @@ def flatten(g, input, start_dim, end_dim):
         return _unimplemented("flatten",
                               "input size not accessible "
                               "(consider using reshape op instead of flatten op to export to ONNX)")
-    input_dims = input.type().sizes()
-    output_dims = []
-    for i in range(0, dim):
-        if start_dim < i and end_dim >= i:
-            output_dims[start_dim] = output_dims[start_dim] * input_dims[i]
-        else:
-            output_dims.append(input_dims[i])
-    shape = g.op("Constant", value_t=torch.LongTensor(output_dims))
-    p = _reshape_from_tensor(g, input, shape)
+    input_size = g.op("Shape", input)
+    slice1 = sym_help._slice_helper(g, input_size, axes=[0], starts=[0], ends=[start_dim])
+    slice2 = sym_help._slice_helper(g, input_size, axes=[0],  starts=[start_dim], ends=[end_dim+1])
+    slice2 = g.op("ReduceProd", slice2)
+    slices = [slice1, slice2]
```

Comment:
I'm wondering if we can use `-1` for one of the slices, like `[slice1, torch.tensor([-1])]` and `[slice1, slice2, torch.tensor([-1])]`.

Reply:
I don't think it would work. I'm getting this error: onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Node (Concat_98) Op (Concat) [ShapeInferenceError] All inputs to Concat must have same rank

Reply:
yea i forgot to put the tensor value in Constant node. How about 
`[slice1, g.op("Constant", value_t=torch.tensor([-1], dtype=torch.long))]` 
and 
`[slice1, g.op("Constant", value_t=torch.tensor([-1], dtype=torch.long)), slice3]`?